contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -193,12 +193,11 @@ We extend kubeadm with a set of flags and helper commands for managing and using
 ### `kubeadm join` flags
 
 * `--token` This sets the token for both discovery and bootstrap auth.
-* `--discovery-url` If set this will grab the cluster-info data (a kubeconfig) from a URL. Due to the sensitive nature of this data, we will only support https URLs.  This also supports `username:password@host` syntax for doing HTTP auth.
-* `--discovery-file` If set, this will load the cluster-info from a file.
+* `--discovery-file` If set, this will load the cluster-info from a file on disk or from a HTTPS URL (the HTTPS requirement due to the sensitive nature of the data)
 * `--discovery-token` If set, (or set via `--token`) then we will be using the token scheme described above.
 * `--tls-bootstrap-token` (not officially part of this spec) This sets the token used to temporarily authenticate to the API server in order to submit a CSR for signing.  If `--insecure-experimental-approve-all-kubelet-csrs-for-group` is set to `system:bootstrappers` then these CSRs will be approved automatically for a hands off joining flow.
```

Comment:
A ClusterRoleBinding between a ClusterRole (like these: https://github.com/kubernetes/kubernetes/pull/49284) and a group that the token got authenticated into.